lib/rules/disallow-multiple-var-decl.js

```diff
@@ -76,22 +99,53 @@ module.exports.prototype = {
     check: function(file, errors) {
         var inStrictMode = this.strictMode;
         var exceptUndefined = this.exceptUndefined;
+        var exceptRequire = this.exceptRequire;
 
         file.iterateNodesByType('VariableDeclaration', function(node) {
-            var hasDefinedVariables = node.declarations.some(function(declaration) {
+            var definedVariables = node.declarations.filter(function(declaration) {
                 return !!declaration.init;
             });
+            var hasDefinedVariables = definedVariables.length > 0;
+
+            var requireStatements = node.declarations.filter(function(declaration) {
+                var init = declaration.init;
+
+                return init &&
+                    init.callee &&
+                    init.callee.name === 'require';
+            });
+            var allRequireStatements = requireStatements.length === node.declarations.length;
 
             var isForStatement = node.parentNode.type === 'ForStatement';
 
             // allow single var declarations
-            if (node.declarations.length === 1 ||
-                // allow multiple var declarations in for statement unless we're in strict mode
-                // for (var i = 0, j = myArray.length; i < j; i++) {}
-                !inStrictMode && isForStatement ||
-                // allow multiple var declarations with all undefined variables in exceptUndefined mode
-                // var a, b, c
-                exceptUndefined && !hasDefinedVariables) {
+            if (node.declarations.length === 1) {
+                return;
+            }
+
+            // allow multiple var declarations in for statement unless we're in strict mode
+            // for (var i = 0, j = myArray.length; i < j; i++) {}
+            if (!inStrictMode && isForStatement) {
+                return;
+            }
+
+            // allow multiple var declarations with all undefined variables in exceptUndefined mode
+            // var a, b, c
+            if (exceptUndefined && !hasDefinedVariables) {
+                return;
+            }
+
+            // allow multiple var declaration with all require
+            // var a = require("a"), b = require("b")
+            if (exceptRequire && allRequireStatements) {
+                return;
+            }
+
+            // allow multiple var declarations only with require && undefined
+            // var a = require("a"), b = require("b"), x, y
+            if (exceptUndefined && exceptRequire &&
+                (definedVariables.length - requireStatements.length) === 0 &&
+                (definedVariables.length + requireStatements.length) === node.declarations.length) {
```

Comment:
Not a big deal, but you can just do `definedVariables.length === requireStatements.length` here right?